xcm/xcm-builder/src/origin_conversion.rs

```diff
@@ -321,3 +321,20 @@ where
 		})
 	}
 }
+
+/// `Convert` implementation to convert from an origin which passes the check of an `EnsureOrigin`
+/// into a voice of a given pluralistic `Body`.
+pub struct OriginToPluralityVoice<RuntimeOrigin, EnsureBodyOrigin, Body>(
+	PhantomData<(RuntimeOrigin, EnsureBodyOrigin, Body)>,
+);
+impl<RuntimeOrigin: Clone, EnsureBodyOrigin: EnsureOrigin<RuntimeOrigin>, Body: Get<BodyId>>
+	Convert<RuntimeOrigin, MultiLocation>
+	for OriginToPluralityVoice<RuntimeOrigin, EnsureBodyOrigin, Body>
+{
+	fn convert(o: RuntimeOrigin) -> Result<MultiLocation, RuntimeOrigin> {
+		match EnsureBodyOrigin::try_origin(o) {
+			Ok(_) => Ok(Junction::Plurality { id: Body::get(), part: BodyPart::Voice }.into()),
```

Comment:
So `BodyPart` also contains variants called `Fraction`, `AtLeastProportion` and `MoreThanProportion`. It would look like to me that we should include the results of a vote from the body when it's appropriate, e.g. the fellowship should have a `BodyPart` that tells you exactly how much voting power is given to a particular XCM. That way, the destination could also filter out results that are not passing the majority (or supermajority, whichever the destination decides to be).

Reply:
This would be very flexible, but I'm not sure how *useful* it really would be. Political bodies generally have a single means of enacting their voice, and rarely need myriad ways of determining whether the body supports or rejects a particular assertion.

Reply:
Make sense to me.
But this will require quite a bit of changes. 
I would propose to go simple for now, and as soon as we have a use case for it, we implement it.
What do you think?

The required changes at least are,
Origins should contain an additional info. 
Referenda pallet's submit call, should not receive `PalletsOrigin` as an argument, can be replaced by track id.
The pallet should be able to work with unit and non-unit like enum origins. 
Also with Gov v2, approve and support thresholds depends from a time passed for decision.

Reply:
Hmm yeah, I suppose it does make sense for the origin to decide what constitutes as a single voice on their own; the destination doesn't _really_ need to have a say in the process. We could punt this down the road until we have an actual use case for it.